packages/sanity/src/config/studio/types.ts

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import {Tool} from '../types'
+
+// Components
+export interface LayoutProps {
+  renderDefault: (props: LayoutProps) => React.ReactElement
+}
+
+export interface LogoProps {
+  title: string
+  renderDefault: (props: LogoProps) => React.ReactElement
+}
+
+export interface NavbarProps {
+  renderDefault: (props: NavbarProps) => React.ReactElement
+}
+
+export interface ToolMenuProps {
+  activeToolName?: string
+  closeSidebar: () => void
+  context: 'sidebar' | 'topbar'
+  isSidebarOpen: boolean
+  tools: Tool[]
+  renderDefault: (props: ToolMenuProps) => React.ReactElement
+}
+
+// Config
+export interface StudioComponents {
+  layout: React.ComponentType<Omit<LayoutProps, 'renderDefault'>>
+  logo: React.ComponentType<Omit<LogoProps, 'renderDefault'>>
+  navbar: React.ComponentType<Omit<NavbarProps, 'renderDefault'>>
+  toolMenu: React.ComponentType<Omit<ToolMenuProps, 'renderDefault'>>
+}
+
+export interface StudioComponentsPluginOptions {
+  layout?: React.ComponentType<LayoutProps>
```

Comment:
Could we add some ts-docs to these so they describe a bit about what they will configure?

I am especially interested to know if layout will allow plugins to provide global studio context. Ie: If I wrap layout in a context-provider, can the context be accessed in form-context and also in another part of the studio react tree?

Reply:
@hermanwikner and @rexxars: Would this be an opportunity to test out what a ts-docs workflow could look like?

Reply:
https://github.com/sanity-io/sanity/pull/3720